server/routes/visitors.ts

```diff
@@ -137,6 +141,127 @@ export default function routes(
     }
   )
 
+  router.post(
+    '/select-date-and-time/:offenderNo',
+    body('visit-date-and-time').custom((value: string, { req }) => {
+      const slotsList = req.session.slotsList as VisitSlotList
+
+      // check selected slot is in the list that was shown and has available tables
+      const selectedSlot: VisitSlot = Object.values(slotsList)
```

Comment:
Do you need this with the `.flat()` ? Can't you just do `slotsList.reduce....` ? I'm probably missing something, but `reduce` will give you the value rather than the key anyway?

Reply:
Also would `.filter` work rather than `reduce`?